source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1637 @@
+   ↓-to-S' a = ⌜ S'-equiv-↓ a ⌝⁻¹
+
+   S'-monotone-ish : (x y : ⟨ L ⟩)
+                   → (x ≤ y) holds
+                   → S' x
+                   → S' y
+   S'-monotone-ish x y o =
+    ↓-to-S' y ∘ ↓-monotonicity-lemma ϕ x y o ∘ S'-to-↓ x
+
+  Γ : ⟨ L ⟩ → ⟨ L ⟩
+  Γ a = ⋁ (S' a , β ∘ pr₁ ∘ S'-to-↓ a)
+
+\end{code}
+
+We show that Γ is monotone.
+
+\begin{code}
+
+  Γ-is-monotone : is-monotone-endomap L Γ
+  Γ-is-monotone x y o =
+   ↓-has-sup-implies-monotone ϕ x y (Γ x) (Γ y) o Γx-is-lub Γy-is-lub
+   where
+    Γx-is-lub : (Γ x is-lub-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+    Γx-is-lub = sup-of-small-fam-is-lub L (β ∘ ↓-to-base ϕ x) (i x)      
+    Γy-is-lub : (Γ y is-lub-of (ϕ ↓ y , β ∘ ↓-to-base ϕ y)) holds
+    Γy-is-lub = sup-of-small-fam-is-lub L (β ∘ ↓-to-base ϕ y) (i y)
+
+\end{code}
+
+We now show that every monotone map determines and local inductive definition.
+
+\begin{code}
+
+ monotone-map-give-local-ind-def : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                                 → is-monotone-endomap L f
+                                 → Σ ϕ ꞉ 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩) ,
+                                   Σ i ꞉ (is-local ϕ) ,
+                                    ((x : ⟨ L ⟩) → (Γ ϕ i) x ＝ f x)
+ monotone-map-give-local-ind-def f f-mono = (ϕ , i , H)
+  where
+   ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)
+   ϕ (b , a) =
+    (Lift 𝓤 (b ≤ᴮ f a) , equiv-to-prop (Lift-≃ 𝓤 (b ≤ᴮ f a)) ≤ᴮ-is-prop-valued)
+   ↓ᴮf-equiv-↓-tot : (a : ⟨ L ⟩) → small-↓ᴮ (f a) ≃ ϕ ↓ a
+   ↓ᴮf-equiv-↓-tot a =
+    Σ-cong' (λ z → z ≤ᴮ f a)
+            ((λ z → (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds))
+            ↓ᴮf-equiv-↓
+    where
+     ↓ᴮf-equiv-↓ : (z : B)
+                 → (z ≤ᴮ f a)
+                 ≃ (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+     ↓ᴮf-equiv-↓ z =
+      ⌜ prop-≃-≃-↔ fe ≤ᴮ-is-prop-valued ∥∥-is-prop ⌝⁻¹ (↓ᴮf-to-↓ , ↓-to-↓ᴮf)
+      where
+       ↓ᴮf-to-↓ : z ≤ᴮ f a
+                → (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+       ↓ᴮf-to-↓ o = ∣ (a , ⌜ ≃-Lift 𝓤 (z ≤ᴮ f a) ⌝ o , reflexivity-of L a) ∣
+       ↓-to-↓ᴮf : (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+                → z ≤ᴮ f a
+       ↓-to-↓ᴮf = ∥∥-rec ≤ᴮ-is-prop-valued ↓-to-↓ᴮf'
+        where
+         ↓-to-↓ᴮf' : Σ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds → z ≤ᴮ f a
+         ↓-to-↓ᴮf' (a' , o , r) =
+          ≤-to-≤ᴮ (transitivity-of L (β z) (f a') (f a)
+                                   (≤ᴮ-to-≤
+                                   (⌜ ≃-Lift 𝓤 (z ≤ᴮ f a') ⌝⁻¹ o))
+                                   (f-mono a' a r))
+   i : is-local ϕ
+   i a = (small-↓ᴮ (f a) , ↓ᴮf-equiv-↓-tot a)
+   G : (x : ⟨ L ⟩) → (f x is-lub-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds 
+   G x = (fx-is-upbnd , fx-is-least-upbnd)
+    where
+     fx-is-upbnd : (f x is-an-upper-bound-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+     fx-is-upbnd (b , e) = ↓-to-fx-upbnd e
+      where
+       ↓-to-fx-upbnd : (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ x) holds) holds
```

Comment:
Again, the unwritten convention is to avoid overly descriptive names in `where` bindings, but this is not very important.